### DISCUSSION TITLE
Problem running mpfs-hal-ddr-demo

### DISCUSSION THREAD

**Discussion by andrenemat** [polarfire-soc/polarfire-soc-documentation#286] [Applications and demos]
When I build the project, I get an error:

_In file included from ../src/platform/mpfs_hal/startup_gcc/newlib_stubs.c:21:
../src/platform/mpfs_hal/startup_gcc/../mss_hal.h:31:10: fatal error: common/mss_legacy_defines.h: No such file or directory
 #include "common/mss_legacy_defines.h"_

Inside the mss_hal.h, I see it is calling a "common/mss_legacy_defines.h"  but cannot find it. 

Can you please help figure out what might be the issue?

**Comment by TommyMurphyTM1234** [comment]
I cloned `https://github.com/polarfire-soc/polarfire-soc-bare-metal-examples` and then imported the `https://github.com/polarfire-soc/polarfire-soc-bare-metal-examples/tree/main/driver-examples/mss/mpfs-hal/mpfs-hal-ddr-demo` example into SoftConsole v2022.2-RISC-V-747 and then right clicked on the `mpfs-hal-ddr-demo` project and selected _Build Configurations > Build All_ and all configured build configurations built correctly without any errors/problems.

![2023-02-04 00 45 50](https://user-images.githubusercontent.com/14550604/216735934-a31cc1bf-037c-4b3d-9d50-00ce40036c27.png)

You probably need to post more detailed information about what exactly you did to arrive at a situation whereby you experienced problems while compiling the project.

**Reply by andrenemat** [reply]
Thank you Tommy,

Got it working. I think the problem was that I was using Import Project from file or archive. I cleaned up everything and tried Import into Workspace and that worked.
